### PR TITLE
add checklist to pr

### PR DESCRIPTION
**Pull request by SallyMcGrath**
We are encouraging new and first timer contributors this year so I have updated this checklist to help them put their best foot forward.

### CODE REVIEW

**Comment by SallyMcGrath** [comment]
Note, suggest also https://github.com/kentaro-m/task-completed-checker-action

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -3,6 +3,14 @@
 Module:
 Week(s):
 
+## Checklist
+
+- [ ] I have read the [contributing guidelines](https://syllabus.codeyourfuture.io/contributing/overview)
+- [ ] I have checked my spelling and grammar with an [automated tool](https://www.grammarly.com/grammar-check)
+- [ ] I have previewed my changes to check the [markdown renders](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) as I intend
+- [ ] I have run my code to check it compiles
```
Comment (SallyMcGrath) [inline .github/pull_request_template.md]:
```suggestion
- [ ] I have run my code to check it works
```

**Comment by Dedekind561** [comment]
LGTM

**Review by Dedekind561 — APPROVED** [review]
LGTM

**Pull request merged**